src/System.Net.Requests/src/System/Net/TimerThread.cs

```diff
@@ -190,33 +153,37 @@ internal static Queue GetOrCreateQueue(int durationMilliseconds)
             }
 
             TimerQueue queue;
-            WeakReference weakQueue = (WeakReference)s_QueuesCache[durationMilliseconds];
+            object key = durationMilliseconds; // Box once.
+            WeakReference weakQueue = (WeakReference)s_queuesCache[key];
             if (weakQueue == null || (queue = (TimerQueue)weakQueue.Target) == null)
             {
-                lock (s_NewQueues)
+                lock (s_newQueues)
                 {
-                    weakQueue = (WeakReference)s_QueuesCache[durationMilliseconds];
+                    weakQueue = (WeakReference)s_queuesCache[key];
                     if (weakQueue == null || (queue = (TimerQueue)weakQueue.Target) == null)
                     {
                         queue = new TimerQueue(durationMilliseconds);
                         weakQueue = new WeakReference(queue);
-                        s_NewQueues.AddLast(weakQueue);
-                        s_QueuesCache[durationMilliseconds] = weakQueue;
+                        s_newQueues.AddLast(weakQueue);
+                        s_queuesCache[key] = weakQueue;
 
                         // Take advantage of this lock to periodically scan the table for garbage.
-                        if (++s_CacheScanIteration % c_CacheScanPerIterations == 0)
+                        if (++s_cacheScanIteration % CacheScanPerIterations == 0)
                         {
-                            List<int> garbage = new List<int>();
-                            foreach (DictionaryEntry pair in s_QueuesCache)
+                            var garbage = new List<object>();
```

Comment:
Not necessarily as part of this PR, but I wonder if there'd be benefit to lazily-allocating this only if there's something found to clean up.